### PR TITLE
Address issue #7 and issue #10

### PR DESCRIPTION
**Pull request by supervaka**
Issue #10 seems to be fully resolved.
For issue #7 there is still a bug.
LED ring behavior still changes depending on HID mode or not, details are in the commit history.

still thinking about how to better organize the code

### CODE REVIEW

File: fw/ws2812.c [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+// Setleds for standard RGB
+void inline ws2812_setleds(struct cRGB *ledarray, uint16_t leds)
+{
+   ws2812_setleds_pin(ledarray,leds, _BV(ws2812_pin));
+}
+
+void inline ws2812_setleds_pin(struct cRGB *ledarray, uint16_t leds, uint8_t pinmask)
+{
+  ws2812_sendarray_mask((uint8_t*)ledarray,leds+leds+leds,pinmask);
+  _delay_us(ws2812_resettime);
```
Comment (HWXLR8) [inline fw/ws2812.c:31]: do you think this would cause any issues with the asynchronous nature of this program?

**Review by HWXLR8 — CHANGES_REQUESTED** [review]
Absolutely fantastic work. Really glad to see this working.

File: fw/analog_turntable.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
+#include "timer.h"
+
+typedef struct {
+  // config
+
+  // Number of ticks we need to advance before recognizing an input
+  uint32_t deadzone;
+  // How long to sustain the input before clearing it (if opposite direction is input, we'll release immediately)
+  uint32_t sustain_ms;
+  // Always provide a zero-input for one poll before reversing?
```
[the remark this answers is not in the document]
Comment (supervaka) [reply fw/analog_turntable.h:10]: for 
```c
// Always provide a zero-input for one poll before reversing?
bool clear;
```
when `clear == true`, the answer to the question is yes
so we are "clearing" the input with a zero-input before reversing the turntable direction

File: fw/ws2812.c (hunk 1 above)
Comment (supervaka) [reply fw/ws2812.c:31]: no idea
FWIW `ws2812_resettime` is set to be 50 microseconds (so 0.05 milliseconds I believe)

File: fw/beef.c [hunk 3]
```diff
@@ -80,6 +119,28 @@ int main(void) {
   }
 }
 
+// this refers to the hardware timer peripheral
+// unrelated to the timer class in timer.h
+// https://exploreembedded.com/wiki/AVR_Timer_programming
```
[the remark this answers is not in the document]
Comment (supervaka) [reply fw/beef.c:124]: I agree, to disambiguate I prefixed the function name with `hardware` so it is `hardware_timer1_init()`, but what do you think about `counter1_init()` or `hardware_counter1_init()` or do you have any suggestions?

File: fw/beef.c [hunk 4]
```diff
@@ -26,17 +26,46 @@ USB_ClassInfo_HID_Device_t Joystick_HID_Interface = {
...
 // temporary hack? because set_led() needs access to buttons[]
 button_pins* buttons_ptr;
 
+struct cRGB led[RING_LIGHT_LEDS];
+
+// milliseconds initialized before #include 
+// so that analog_turntable.h->timer.h can use milliseconds
+volatile uint32_t milliseconds = 0;
+#include "analog_turntable.h"
+#include "tt_rgb_manager.h"
```
[the remark this answers is not in the document]
Comment (supervaka) [reply fw/beef.c:43]: Maybe I could move line 41 into `timer.h`?

File: fw/analog_turntable.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (HWXLR8) [reply fw/analog_turntable.h:8]: ah, my mistake. That's fine

File: fw/ws2812.c (hunk 1 above)
Comment (HWXLR8) [reply fw/ws2812.c:31]: ok, something to keep in mind for the future that there is some usage of `_delay_us` pulled in by this lib

**Comment by HWXLR8** [comment]
Thanks again, great work!

**Pull request merged**